### ISSUE TITLE
Sentry isn't catching errors in pipelines

### ISSUE THREAD

**Issue by jim** [City-Bureau/city-scrapers#302]
We're having an issue with one of the pipelines that Sentry isn't catching. Investigate if we need to do something to configure it to catch those errors as well as those that occur within spiders.

**Comment by vincecima** [comment]
@jim is there an example of a error or an easy way to replicate locally?

**Comment by jim** [comment]
@vincecima Having any of the pipelines raise an exception would do it. To reproduce the issue, you can add a line to the `process_item ` method in `city_scrapers/pipelines/localExporter.py` to throw an exception.

I looked into this a bit and I think the approach that will work is this:

```diff
diff --git a/city_scrapers/spider.py b/city_scrapers/spider.py
index 3f9392b..217fc57 100644
--- a/city_scrapers/spider.py
+++ b/city_scrapers/spider.py
@@ -29,3 +29,14 @@ class Spider(scrapy.Spider):
         """
         tz = timezone(source_tz)
         return tz.localize(datetime_object)
+
+
+    def _itemproc_finished(self, output, item, response, spider):
+        """ItemProcessor finished for the given ``item`` and returned ``output``
+        """
+        super()._itemproc_finished(output, item, response, spider)
+
+        if isinstance(output, Failure) and not isinstance(ex, DropItem):
+            # NOTIFY SENTRY
```

Scrapy has a [signal system](https://doc.scrapy.org/en/latest/topics/signals.html), but it doesn't have a signal for failures within pipelines unless those pipelines throw a couple special exceptions such as `DropItem`. We want to catch all exceptions and not just the error cases we anticipate, so that's not going to work for us.

**Comment by vincecima** [comment]
@jim thanks for the additional context. I added a sample method (see https://github.com/City-Bureau/city-scrapers/pull/347). When I run `scrapy crawl chi_police` I don't see any of my logging output.

I get that we are overriding the Spider implementation of `_itemproc_finished`, but I don't see where most of the spiders actually use that Spider as a superclass. I am a Python novice, so I might be missing something obvious.

**Comment by jim** [comment]
@vincecima The scrapers _should_ be inheriting from our Spider subclass [like chi_animal does](https://github.com/City-Bureau/city-scrapers/blob/master/city_scrapers/spiders/chi_animal.py#L7-L10):

```python
from city_scrapers.spider import Spider

class Chi_animalSpider(Spider):
    name = 'chi_animal'
```

Perhaps my grand scheme to override this method won't work for some other reason.

**Comment by vincecima** [comment]
Thanks @jim, I am unfamiliar with Python's class syntax, so that escaped me. Will take a deeper look today on why the method isn't getting called as expected.

**Comment by jim** [comment]
It turns out I mixed up the `Spider` and the `Scraper` classes used internally by Scrapy, so my idea isn't going to work 😦

Feel free to dig into this more, or if you have lost interest at this point I'm happy to take a swing as well.

**Comment by pjsier** [comment]
We should also consider throwing errors at  404 response codes as well, just making a note

**Comment by pjsier** [comment]
@jim I took another look at this, and it looks like the problem might be that we're just setting the exit code rather than throwing any exceptions [in `crawl.py` here](https://github.com/City-Bureau/city-scrapers/blob/master/city_scrapers/commands/crawl.py#L22).

When I run it locally with an exception, the process doesn't fail out, but when I raise a `ValueError` in that line instead of setting the exit code it does. Also, we could potentially set [`CLOSESPIDER_ERRORCOUNT`](https://doc.scrapy.org/en/latest/topics/extensions.html#closespider-errorcount) if we want them to end after the first problem.

**Comment by pjsier** [comment]
Closed by #428
